### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Bogdan-Zinovij**

### CODE REVIEW

File: server/.env.example [hunk 1]
```diff
@@ -1,2 +1,4 @@
 EXPRESS_PORT=8080
-HOST=localhost
\ No newline at end of file
+HOST=localhost
+tokenSecret=16f2c5s
```
Comment (artemmatiushenko1) [inline server/.env.example:3]: Use only one letters case for naming env variables

File: server/src/app.js [hunk 2]
```diff
@@ -2,10 +2,10 @@
 
 const express = require('express');
 const app = express();
+const authRouter = require('./routers/authRouter');
+const PREFIX = '/api/v1'
```
Comment (KhrapkoVasyl) [inline server/src/app.js:6]: Move constant PREFIX to config.js

File: server/src/middleware/authMiddleware.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const jwt = require('jsonwebtoken');
+
+module.exports = function (req, res, next) {
+  if (req.method === 'OPTIONS') {
```
Comment (artemmatiushenko1) [inline server/src/middleware/authMiddleware.js:6]: Magic string 'OPTIONS'

**Review by artemmatiushenko1 — CHANGES_REQUESTED** [review]

File: server/src/controllers/authController.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+const createUserService = require('../services/createUserService');
+const comparePasswordsService = require("../services/comparePasswordsService");
+const findUserByNameService = require("../services/findUserByNameService")
+
+class AuthController {
+  async registration(req, res) {
+    try {
+      const errors = validationResult(req);
+      if (!errors.isEmpty()) {
+        return res.status(400).json({ message: errors });
```
Comment (KhrapkoVasyl) [inline server/src/controllers/authController.js:14]: return json must follow the form:
{
status: 'success' || 'failed',
message: 'message' || err.message
}

File: server/src/controllers/authController.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      const User = await findUserByNameService(login);
+
+      const isPasswordValid = await comparePasswordsService(login, password);
+      if (!isPasswordValid) {
+        throw new Error('Incorrect password!');
+      }
+
+      const accessToken = generateAccessToken(User.id);
+
+      res.status(200).json({ message: 'Successfully logged in!', accessToken });
```
Comment (KhrapkoVasyl) [inline server/src/controllers/authController.js:47]: incorrect json form

File: server/src/routers/authRouter.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const Router = require('express');
+const authRouter = new Router();
+const authController = require('../controllers/authController');
+const {
+  registrValidationScheme,
```
Comment (KhrapkoVasyl) [inline server/src/routers/authRouter.js:7]: It'd be better to use name signup and signin

File: server/src/controllers/authController.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const generateAccessToken = require('../utils/generateAccessToken');
+const { validationResult } = require('express-validator');
+const createUserService = require('../services/createUserService');
+const comparePasswordsService = require("../services/comparePasswordsService");
+const findUserByNameService = require("../services/findUserByNameService")
+
+class AuthController {
+  async registration(req, res) {
```
Comment (artemmatiushenko1) [inline server/src/controllers/authController.js:10]: You should better name this method ```signUp```

File: server/src/services/createUserService.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+const db = require('../db/db');
+
+const createUserService = async (login, password) => {
+  const User = await db.findUserByName(login);
```
Comment (KhrapkoVasyl) [inline server/src/services/createUserService.js:6]: use user not User

File: server/src/controllers/authController.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+      const accessToken = generateAccessToken(User.id);
+
+      res.status(201).json({ message: 'User was created!', accessToken });
+    } catch (err) {
+      res.status(400).json({ message: err.message });
+    }
+  }
+
+  async login(req, res) {
```
Comment (artemmatiushenko1) [inline server/src/controllers/authController.js:29]: You should better name this method ```signIn```

**Review by artemmatiushenko1 — CHANGES_REQUESTED** [review]

File: server/src/services/createUserService.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+  if (User) {
+    throw new Error('User with the same name is already exists!');
+  }
+
+  const newUser = await db.insertUser({ name: login, password });
+
+  if (!newUser) {
+    throw new Error('Failed to create User!');
```
Comment (KhrapkoVasyl) [inline server/src/services/createUserService.js:14]: do please 'user', not 'User'

File: server/src/services/findUserByNameService.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+const db = require('../db/db');
+
+const findUserByNameService = async (login) => {
+  const User = await db.findUserByName(login);
```
Comment (KhrapkoVasyl) [inline server/src/services/findUserByNameService.js:6]: user

File: server/.env.example [hunk 12]
```diff
@@ -1,3 +1,5 @@
 EXPRESS_PORT=8080
 HOST=localhost
+tokenSecret=16f2c5s
```
Comment (KhrapkoVasyl) [inline server/.env.example:3]: Better: TOKEN_SECRET

File: server/.env.example [hunk 13]
```diff
@@ -1,3 +1,5 @@
...
+sessionDuration=15m
```
Comment (KhrapkoVasyl) [inline server/.env.example:4]: SESSION_DURATION

**Review by KhrapkoVasyl — CHANGES_REQUESTED** [review]

**Review by artemmatiushenko1 — APPROVED** [review]

**Pull request merged**